### PR TITLE
docs: update Kafka ACL docs

### PR DESCRIPTION
**Pull request by staceysalamon-aiven**
## About this change—what it does

Updates descriptions for Aiven ACLs and Kafka-native ACLs to provide clarification on when to use each. Adds example usage for Kafka-native ACLs. Updates field descriptions for native ACLs.

### CODE REVIEW

**Review by roope-kar — APPROVED** [review]
lgtm, @harshini-rangaswamy do you want to take a look?

File: docs/resources/kafka_native_acl.md [hunk 1]
```diff
@@ -3,32 +3,51 @@
...
 - `pattern_type` (String) Resource pattern used to match specified resources. The possible values are `LITERAL` and `PREFIXED`. Changing this property forces recreation of the resource.
-- `permission_type` (String) The permission type. The possible values are `ALLOW` and `DENY`. Changing this property forces recreation of the resource.
-- `principal` (String) Principal is in type:name' format. Maximum length: `256`. Changing this property forces recreation of the resource.
+- `permission_type` (String) Specifies whether the action is explicitly allowed or denied for the principal on the specified resource. The possible values are `ALLOW` and `DENY`. Changing this property forces recreation of the resource.
+- `principal` (String) Identities in `type:name` format that the permissions apply to. Maximum length: `256`. Changing this property forces recreation of the resource.
```
Comment (harshini-rangaswamy) [inline docs/resources/kafka_native_acl.md:42]: Only `User:` is supported for the `principal`, and wildcards (for example, `User:alice*`) can be used to match usernames.
cc: @roope-kar

Comment (staceysalamon-aiven) [reply docs/resources/kafka_native_acl.md]: Updated.

**Comment by harshini-rangaswamy** [comment]
> lgtm, @harshini-rangaswamy do you want to take a look?

It looks good to me, too. Thanks, @staceysalamon-aiven!

**Review by roope-kar — APPROVED** [review]

**Review by roope-kar — APPROVED** [review]

**Review by byashimov — APPROVED** [review]

**Pull request merged**